### PR TITLE
chore: support windows vhd build for 23H2

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Support windows vhd build for 23H2.
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: .pipelines/e2e-windows.yaml [hunk 1]
```diff
@@ -49,6 +49,14 @@ parameters:
   displayName: Windows NV Series VM Size (gen2)
   type: string
   default: Standard_NV6ads_A10_v5
+- name: test23H2containerd
```
Comment (ShiqianTao) [inline .pipelines/e2e-windows.yaml:52]: I prefer to remove "containerd" that same as other varibales

Comment (junjiezhang1997) [reply .pipelines/e2e-windows.yaml]: Fixed. Miss here before.

File: .pipelines/e2e-windows.yaml [hunk 2]
```diff
@@ -116,4 +124,30 @@ stages:
...
+  - template: ./templates/e2e-windows-template.yaml
+    parameters:
+      windowsImage: 23H2
+      windowsOSSKU: WindowsAnnual
+      windowsVMSize: Standard_D2_v2
+      windowsDistro: aks-windows-23H2
+      windowsNodeImageVersion: AKSWindows-23H2-test-2023.02.07
+      storageAccount: akswinstore23H2
+
+- stage: test_23H2_gen2
```
Comment (AbelHu) [inline .pipelines/e2e-windows.yaml:142]: Not an issue. Maybe we can keep only one test (default gen1 and later change it to gen2) to save the cost

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR

File: .pipelines/e2e-windows.yaml (hunk 2 above)
Comment (junjiezhang1997) [reply .pipelines/e2e-windows.yaml:142]: Create a task to track this issue.

**Pull request merged**